linera-sdk/src/service/mod.rs

```diff
@@ -24,14 +23,20 @@ macro_rules! service {
     ($application:ty) => {
         #[doc(hidden)]
         #[no_mangle]
-        fn __service_handle_query(
-            argument: Vec<u8>,
-        ) -> Result<Vec<u8>, String> {
-            $crate::service::run_async_entrypoint(
-                <
-                    <$application as $crate::Service>::Storage as $crate::ServiceStateStorage
-                >::handle_query(argument),
-            )
+        fn __service_handle_query(argument: Vec<u8>) -> Result<Vec<u8>, String> {
+            let request = serde_json::from_slice(&argument).map_err(|error| error.to_string())?;
```

Comment:
I missed where that JSON deserialization was before the change… also, I thought applications could choose whether their queries are JSON or something else?

Reply:
Apparently not, because the `ServiceStateStorage` always used JSON:

https://github.com/linera-io/linera-protocol/blob/a155251fdbe9c36603e0f5b3f92ae77eb159d342/linera-sdk/src/service/storage.rs#L48-L54
https://github.com/linera-io/linera-protocol/blob/a155251fdbe9c36603e0f5b3f92ae77eb159d342/linera-sdk/src/service/storage.rs#L71-L75